tensorflow_lite_support/python/task/vision/image_segmenter.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image segmenter task."""
+
+import dataclasses
+import numpy as np
+from typing import List, Tuple, Optional
+
+from tensorflow_lite_support.python.task.core.proto import base_options_pb2
+from tensorflow_lite_support.python.task.processor.proto import segmentation_options_pb2
+from tensorflow_lite_support.python.task.processor.proto import segmentations_pb2
+from tensorflow_lite_support.python.task.vision.core import tensor_image
+from tensorflow_lite_support.python.task.vision.core.pybinds import image_utils
+from tensorflow_lite_support.python.task.vision.pybinds import _pywrap_image_segmenter
+
+_OutputType = segmentation_options_pb2.OutputType
+_CppImageSegmenter = _pywrap_image_segmenter.ImageSegmenter
+_BaseOptions = base_options_pb2.BaseOptions
+
+
+@dataclasses.dataclass
+class ColoredLabel:
+  label: str
+  """The label name."""
+
+  color: Tuple[int, int, int]
+  """The RGB representation of the label's color."""
+
+
+@dataclasses.dataclass
```

Comment:
@kinaryml You should be using the protos directly instead of redefining them in Python.

Reply:
@khanhlvg The Java API seems to be using a class for `ColoredLabel` in order to postprocess the Native C++ segmentation output [here](https://github.com/khanhlvg/tflite-support/blob/master/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/vision/segmenter/ColoredLabel.java) though. WDYT?

Also there's the `Segmentation` class over [here](https://github.com/khanhlvg/tflite-support/blob/121beb1b25f48d5b63581bcdac4cccad82a9d7d5/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/vision/segmenter/Segmentation.java#L81)

Reply:
@khanhlvg Do you think we should delegate [postprocessing](https://github.com/khanhlvg/tflite-support/blob/0494b6408161694d541a2f4c3991fee748a20385/tensorflow_lite_support/python/task/vision/image_segmenter.py#L154) of the masks to the tests and just return the raw segmentation output [here](https://github.com/khanhlvg/tflite-support/blob/0494b6408161694d541a2f4c3991fee748a20385/tensorflow_lite_support/python/task/vision/image_segmenter.py#L136). I did this earlier while implementing the API but let me know which approach sounds better.

Reply:
The Java API seems to post-process the masks and colored labels like [this](https://github.com/khanhlvg/tflite-support/blob/121beb1b25f48d5b63581bcdac4cccad82a9d7d5/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/vision/segmenter/ImageSegmenter.java#L386)

Reply:
Thanks for the clarification. Let's keep your implementation.

Reply:
@khanhlvg I think to keep the implementation consistent with the other Task APIs, we'd have to let `ImageSegmenter` just return `SegmentationResult` just like how the other APIs return `ClassificationResult`, etc., and post-process the segmentation result in the tests. I've applied these changes now.